server/lib/src/idm/account.rs

```diff
@@ -235,7 +232,8 @@ impl Account {
         value: &Entry<EntrySealed, EntryCommitted>,
         qs: &mut QueryServerReadTransaction,
     ) -> Result<Self, OperationError> {
-        let (groups, unix_groups) = load_all_groups_from_account_entry(value, qs)?;
+        let groups = Group::<()>::try_from_account(value, qs)?;
+        let unix_groups = Group::<Unix>::try_from_account(value, qs).unwrap_or_default();
```

Comment:
I do dislike that doing it properly requires this double query and iteration of the groups. If we believe this may become a performance bottleneck or just want it optimized I can write a helper function to replicate the previous single iteration behavior.

Reply:
I'd look at doing that, especially since that call's in the path for oauth2/unix/ldap auth.